crates/e2e-testing/src/testcase.rs

```diff
@@ -146,11 +147,24 @@ impl TestCase {
             spin::registry_push(&appname, registry_app_url.as_str())?;
         }
 
+        // create temp dir as state dir
+        let tempdir = TempDir::new().unwrap().into_path();
```

Comment:
`into_path` means it will no longer clean up the tempdir - it "throws away" the temporary nature and gives you a plain path somewhere in temp space; and dropping that plain path no longer does anything.  (https://docs.rs/tempdir/latest/tempdir/struct.TempDir.html#method.into_path)

You need `tempdir` (the variable you hold on to and drop at the end) to be an _actual `TempDir` instance_.  And you _must_ use `path()` rather than `into_path()` to get the OS path.  (https://docs.rs/tempdir/latest/tempdir/struct.TempDir.html#method.path)  `path()` does not consume the `TempDir` so the `TempDir` continues to "own" the directory and is able to delete it on drop.

I'm sorry, I should have flagged this nuance in my comment suggesting this pattern.  It's a trap we've fallen into before!

Reply:
no you are fine. it's me. I should have tested it more and read documentation about it. fix coming up.